src/help/utils/apiSnippets.tsx

```diff
@@ -0,0 +1,307 @@
+    ? `
+params = ${JSON.stringify(
+        Object.fromEntries(searchParams),
+        (key, value) => {
+          if (key === 'fields') {
+            return value.split(',');
+          }
+          return value;
+        },
+        2
+      )}`
+    : ''
+}
+headers = ${JSON.stringify(Object.fromEntries(headers.entries()), null, 2)}
+base_url = "${url.origin + url.pathname}"
+
+response = requests.get(base_url, headers=headers, params=params)
+if not r.ok:
+  r.raise_for_status()
+  sys.exit()
+
+data = response.${isJSON ? 'json()' : 'text'}
+
+print(${isJSON ? 'json.dumps(data, indent=2)' : 'data'})
+`;
+    },
+  },
+};
+
+// JAVASCRIPT
+const SnippetGeneratorNodeJsPlugin: SnippetGeneratorPlugin = {
+  fn: {
+    // use `requestSnippetGenerator_` + key from config (node_native) for generator fn
+    requestSnippetGenerator_js_fetch: (request) => {
+      const { url, searchParams, method, headers, body, isJSON } =
+        requestExtract(request);
+
+      if (method === 'POST') {
+        return '// No automatic snippet, work in progress';
+      }
+
+      return `// requirement: NodeJS >= 14.8.0, or Firefox, Chrome, or Edge >= 89
+${
+  searchParams
+    ? `
+const searchParams = new URLSearchParams(${JSON.stringify(
+        Object.fromEntries(searchParams),
+        (key, value) => {
+          if (key === 'fields') {
+            return value.split(',');
+          }
+          return value;
+        },
+        2
+      )});`
+    : ''
+}
+const url = \`${url.origin + url.pathname}${searchParams ? `?$\{searchParams}` : ''}\`;
+
+const response = await fetch(url, {
+  headers: ${indent(JSON.stringify(Object.fromEntries(headers.entries()), null, 2))},${
+    method === 'GET'
+      ? ''
+      : `
+  method: "${method}",`
+  }${
+    body
+      ? `
+  body: new FormData(${indent(JSON.stringify(body, null, 2))}),`
+      : ''
+  }
+});
+if (!response.ok) {
+  throw new Error(\`$\{response.status}: $\{response.statusText}\`);
+}
+
+const data = await response.${isJSON ? 'json' : 'text'}();
+
+console.log(data);
+`;
+    },
+  },
+};
+
+// PERL
+const SnippetGeneratorPerlPlugin: SnippetGeneratorPlugin = {
+  fn: {
+    // use `requestSnippetGenerator_` + key from config (node_native) for generator fn
+    requestSnippetGenerator_perl: (request) => {
+      const { url, method, headers, isJSON } = requestExtract(request);
+
+      if (method === 'POST') {
+        return '# No automatic snippet, work in progress';
+      }
+
+      return `# Perl not recommended, limited support
+use strict;
+use warnings;
+use HTTP::Tiny;${isJSON ? '\nuse JSON;' : ''}
+
+my $http = HTTP::Tiny->new();
+
+my $url = '${url}';
+my $response = $http->get($url, {
+    headers => {
+${Array.from(headers.entries())
+  .map(
+    ([key, value], index) =>
+      `${index === 0 ? '' : ',\n'}        '${key}' = '${value}'`
+  )
+  .join('')}
+    }
+});
+
+if (!$response->{success}) {
+    die "Error " . $response->{status} . ": " . $response->{reason};
+}
```

Comment:
I found this useful to understand why it wasn't working:
```suggestion
if (!$response->{success}) {
    print "Error " . $response->{status} . ": " . $response->{reason} . "\n";
    print "Response content: " . $response->{content} . "\n";
    die;
}
```